src/Libraries/RevitServices/Events/EventHandlerProxy.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading;
+using Autodesk.Revit.DB.Events;
+using Autodesk.Revit.UI.Events;
+using Dynamo.Models;
+
+namespace RevitServices.EventHandler
+{
+    /// <summary>
+    /// This is a event handler proxy class to serve as a proxy between the event publisher and
+    /// the event subscriber
+    /// </summary>
+    public class EventHandlerProxy
```

Comment:
@ke-yu 
Thanks for the comments!

But a class won't have a lifetime. I assume you are referring to the class instance. But when creating and releasing the instance(by setting it to null), we have ensured that the lifetime of this instance is within the lifetime of the addin application, that is also why the instance is member of the addin application. As for your worry about that it can be released somewhere, there is not a set method for the EventHandlerProxy property and so it is impossible to set the proxy to null outside the addin application class.

The reason to put EventHandlerProxy to a separate module is for modulization so that the code is more easier to read. Anyway putting EventHandlerProxy as an inner class can not prevent it from being used, we have to make the EventHandlerProxy as public and the class to be public as well. Then the class can still be used like:
var proxy = new DynamoRevitApp.EventHandlerProxy();